macos/ONLYOFFICE/update/updates/ONLYOFFICE/changes/8.1.0/changes.html

```diff
@@ -91,46 +91,83 @@ <h2>New Features</h2>
+            <li>New functions: <code>GETPIVOTDATA</code>, <code>IMPORTRANGE</code></li>
+            <li>New function category: Custom based on jsdoc</li>
+            <li>Version history update: edited cells are highlighted</li>
+            <li>Users get custom protected range cells viewing rights</li>
+            <li>Implemented the ability to copy/move sheets between books in one browser</li>
+            <li>Changed the appearance of the sheet list in the embedded viewer in accordance with the styles of the main spreadsheet editor</li>
         </ul>
         <h4>Presentation Editor</h4>
         <ul>
-            <li>TODO</li>
+            <li>New slide settings on the right panel: show background graphics, reset background to the theme background, apply settings to all slides</li>
+            <li>Added <code>Animation</code> pane</li>
+            <li>Added a mode for editing master slides and templates</li>
         </ul>
         <h4>Forms</h4>
         <ul>
-            <li>TODO</li>
+            <li>When adding a fixed form, now it is inserted without wrapping in front of the text</li>
+            <li>The color of the  Picture placeholder corresponds to the color of the role for this form</li>
+            <li>The thickness of the frame for required fields is now 2px with any zoom</li>
+            <li>Added a button to switch to the editing mode (similar to the button in the editor header) for forms opened in the View or Fill forms mode</li>
+            <li>The presence of this button depends on the integrator's subscription to the onRequestEditRights event and the permissions.edit rights (in case of a viewer)</li>
+        </ul>
+        <h4>PDF Editor</h4>
+        <ul>
+            <li>Added buttons for switching editing/viewing (annotations) modes to the toolbar and the editor header</li>
+            <li>Added tooltips for new or updated functionality (displayed when loading the editor or when switching to the corresponding tab)</li>
+            <li>In the Edit mode, it's possible to add various objects (using the <code>Home</code> and <code>Insert</code> tabs) and configure them using the right panel and context menu</li>
+            <li>Added the ability to add, delete or rotate pages using the context menu and the toolbar</li>
+            <li>Added a mini toolbar for adding annotations when selecting text</li>
         </ul>
 
         <h2>Fixes</h2>
 
-        <h4>All Editors</h4>
-        <ul>
-            <li>TODO</li>
-        </ul>
-        <h4>Document Editor</h4>
-        <ul>
-            <li>TODO</li>
-        </ul>
-        <h4>Spreadsheet Editor</h4>
-        <ul>
-            <li>TODO</li>
-        </ul>
-        <h4>Presentation Editor</h4>
-        <ul>
-            <li>TODO</li>
-        </ul>
-        <h4>Forms</h4>
+        <h4>Security</h4>
         <ul>
-            <li>TODO</li>
+            <li>Fixed the vulnerability in the fs.folderPath field which allows accessing the file system</li>
```

Comment:
Actual in DocumentServer only since the issue relates to external storage.

Reply:
removed